docs/advanced-guides/testing/testing-with-react-testing-library.md

```diff
@@ -133,36 +134,34 @@ Let's say that we've built a redirect component that deals with re-routing the u
 `RedirectHandler.js`
 ```jsx
 import React from 'react'
-import {Route, Switch, Redirect} from 'react-router-dom'
+import { Redirect } from 'react-router-dom'
 
-const RedirectHandler = props => (
-  <Switch>
-    <Redirect from="/the-old-thing" to="the-new-thing">
-  </Switch>
-)
+const RedirectHandler = () => (<Redirect from="/the-old-thing" to="the-new-thing"></Redirect>);
 
-export default RedirectHandler
+export default RedirectHandler;
 ```
 
 And here we have one method of testing that the redirect is behaving as expected.
 
 `RedirectHandler.test.js`
 ```jsx
-import React from 'react'
-import {MemoryRouter, Route} from 'react-router-dom'
-import { render, fireEvent, screen } from '@testing-library/react';
-import RedirectHandler from './RedirectHandler'
+import React from "react";
+import { BrowserRouter } from "react-router-dom";
+import { render } from "@testing-library/react";
+import RedirectHandler from "./RedirectHandler";
+
+it("redirects /the-old-thing to /the-new-thing", () => {
+  window.history.pushState({}, "The Old Thing", "/the-new-thing");
```

Comment:
Do we want to `pushState()` to `/the-new-thing` at the beginning of the test? It seems to me that if this is how the history is set up, then we aren't really testing whether the redirect is happening.

Am I misunderstanding how `pushState` is functioning here?